include/swift/AST/DiagnosticsClangImporter.def

```diff
@@ -157,17 +157,22 @@ NOTE(reference_passed_by_value, none, "function uses foreign reference type "
                                       "'import_reference' contract (outlined in "
                                       "C++ Interop User Manual).",
      (StringRef, StringRef))
+
 NOTE(record_not_automatically_importable, none, "record '%0' is not "
                                                 "automatically importable: %1. "
                                                 "Refer to the C++ Interop User "
                                                 "Manual to classify this type.",
      (StringRef, StringRef))
-NOTE(projection_not_imported, none, "C++ method '%0' that returns unsafe "
-                                    "projection of type '%1' not imported",
-     (StringRef, StringRef))
-NOTE(dont_use_iterator_api, none, "C++ method '%0' that returns an unsafe "
-                                  "iterator not imported: use Swift Sequence "
-                                  "APIs instead",
+
+NOTE(projection_not_imported, none, "C++ method '%0' is unavailable in Swift "
+                                    "as it may return or store an address of a memory location "
```

Comment:
I think "or store" is a confusing way to put this. Maybe "as it returns an address or struct containing an address"? I also don't think you need to say memory location; "address" is sufficient (in my opinion).

Reply:
Ok.  Also, now that the method calls might adopt borrow semantics in the future, I think we can rephrase the error message as 
"C++ method '%0' is unavailable in Swift as it may return an address or a struct containing an address pointing into the receiver instance, which could be destroyed before return value is used"